templates/release.yml

```
@@ -0,0 +1,66 @@
+name: Publish Module
+
+on: workflow_dispatch
+  
+jobs:
+  publish-module:
+    name: Tag Release and Publish to Forge
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.ref }}
+          clean: true
+          fetch-depth: 0
+      - name: Get Version
+        id: gv
+        run: |
+          echo "::set-output name=ver::$(jq --raw-output .version metadata.json)"
+      - name: Install the PDK
+        run: |
+          source /etc/os-release
+          wget https://apt.puppet.com/puppet-tools-release-${UBUNTU_CODENAME}.deb
+          sudo dpkg -i puppet-tools-release-${UBUNTU_CODENAME}.deb
+          sudo apt-get update
+          sudo apt-get install pdk
+      - name: Validate Docs
+        run: |
+          documentation=$(pdk bundle exec puppet strings generate --format markdown)
+          if [ $(echo $documentation | grep -Ec "[1-9]+ undocumented|warn") -gt 0 ]; then
+            echo "Please resolve documentation issues detected below:"
+            echo $documentation
+            exit 1
+          fi
+          if output=$(git status --porcelain) && [ ! -z "$output" ]; then
+            echo "The following changes were detected:"
+            git --no-pager diff REFERENCE.md
+            echo "Please submit a PR of changes after running 'pdk bundle exec puppet strings generate --format markdown'"
+            exit 1
+          fi
+      - name: Validate Changelog
+        run: |
+          pdk bundle exec rake changelog
+          grep '### UNCATEGORIZED PRS; LABEL THEM ON GITHUB' CHANGELOG.md
+          if [ $? -eq 0 ]; then
+            echo "Uncategorized PRs found. Please address missing labels in the PRs below:"
+            git --no-pager diff CHANGELOG.md
+            exit 1
+          fi
```

I think a cleaner way to do this would be:
```suggestion
          grep -c '### UNCATEGORIZED PRS; LABEL THEM ON GITHUB' CHANGELOG.md
          if [ $? -gt 0 ]; then
            echo "Uncategorized PRs found. Please address missing labels in the PRs below:"
            git --no-pager diff CHANGELOG.md
            exit 1
          fi
```

---

Good catch, thanks